src/scenic/core/scenarios.py

```diff
@@ -1,518 +1,595 @@
+                # can see the instance.
+                if vi._observingEntity is not None:
+                    observing_entity = sample[vi._observingEntity]
+                    occluding_objects = tuple(sample[obj] for obj in objects \
+                                         if sample[obj] is not observing_entity \
+                                         and sample[obj] is not vi and obj.occluding)
+                    if not observing_entity.canSee(vi, occludingObjects=occluding_objects):
+                        rejection = 'instance visibility (from observing entity)'
+                        break
+                # Require that the non-observing entity, if one has been specified,
+                # can see the instance.
+                if vi._nonObservingEntity is not None:
+                    non_observing_entity = sample[vi._nonObservingEntity]
+                    occluding_objects = tuple(sample[obj] for obj in objects \
+                                         if sample[obj] is not non_observing_entity \
+                                         and sample[obj] is not vi and obj.occluding)
+                    if non_observing_entity.canSee(vi, occludingObjects=occluding_objects):
+                        rejection = 'instance visibility (from non-observing entity)'
+
+            # Check built-in requirements for objects
+            for i in range(len(objects)):
+                vi = sample[objects[i]]
+                # Require object to be contained in the workspace/valid region
+                container = self.containerOfObject(vi)
+                if not container.containsObject(vi):
+                    rejection = 'object containment'
+                    break
+                # Require object to be visible from the ego object
+                if vi.requireVisible and vi is not ego:
+                    occluding_objects = tuple(sample[obj] for obj in objects if sample[obj] is not ego \
+                                         and sample[obj] is not vi and obj.occluding)
+
+                    if not ego.canSee(vi, occludingObjects=occluding_objects):
+                        rejection = 'object visibility (from ego)'
+                        break
+
+                        break
+                # Require object to not intersect another object
+                if not vi.allowCollisions:
+                    for j in range(i):
+                        vj = sample[objects[j]]
+
+                        if not vj.allowCollisions and vi.intersects(vj):
+                            rejection = 'object intersection'
+                            break
+                if rejection is not None:
+                    break
+            if rejection is not None:
+                optionallyDebugRejection()
+                continue
+            # Check user-specified requirements
+            for req in activeReqs:
+                if req.satisfiedBy(sample) == rv_ltl.B4.FALSE:
```

Comment:
Let's move the comparison into `satisfiedBy`; then only the `requirements` module needs to know about `rv_ltl`, not `scenarios`.